### PR TITLE
Update to account for XContent refactor in 2.x

### PR DESCRIPTION
**Pull request by MaxKsyunz**
### Description
Cherry pick of cd95bc54140efbf8fc60932249f645dc581260e5 plus fixes for 2.x-only changes.
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/1475
 
### Check List
- [x] All tests pass, including unit test, integration test and doctest
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: integ-test/src/test/java/org/opensearch/sql/legacy/RestIntegTestCase.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
 import org.opensearch.client.Request;
 import org.opensearch.client.Response;
 import org.opensearch.common.Strings;
-import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentBuilder;
```
Comment (Yury-Fridlyand) [inline integ-test/src/test/java/org/opensearch/sql/legacy/RestIntegTestCase.java:47]: reorder?

File: integ-test/src/test/java/org/opensearch/sql/legacy/SourceFieldIT.java [hunk 2]
```diff
@@ -15,9 +15,9 @@
 import org.junit.Test;
 import org.opensearch.action.search.SearchResponse;
 import org.opensearch.common.xcontent.LoggingDeprecationHandler;
-import org.opensearch.common.xcontent.NamedXContentRegistry;
 import org.opensearch.common.xcontent.XContentFactory;
-import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
```
Comment (Yury-Fridlyand) [inline integ-test/src/test/java/org/opensearch/sql/legacy/SourceFieldIT.java:19]: reorder?

**Review by Yury-Fridlyand — DISMISSED** [review]
Import reorder may be needed to satisfy the checkstyle.

**Review by rupal-bq — DISMISSED** [review]

File: integ-test/src/test/java/org/opensearch/sql/legacy/SourceFieldIT.java (hunk 2 above)
Comment (derek-ho) [reply integ-test/src/test/java/org/opensearch/sql/legacy/SourceFieldIT.java:19]: +1 on the reorder, is there a linter that can be run?

File: build.gradle [hunk 3]
```diff
@@ -98,6 +98,8 @@ subprojects {
         mavenLocal()
         maven { url "https://aws.oss.sonatype.org/content/repositories/snapshots" }
         mavenCentral()
+        // todo. remove this when lucene 9.4.0 is released
+        maven { url "https://d1nvenhzbhpy0q.cloudfront.net/snapshots/lucene/" }
```
Comment (derek-ho) [inline build.gradle:102]: why was this added?

**Review by derek-ho — APPROVED** [review]
mostly LGTM

File: integ-test/src/test/java/org/opensearch/sql/legacy/SourceFieldIT.java (hunk 2 above)
Comment (Yury-Fridlyand) [reply integ-test/src/test/java/org/opensearch/sql/legacy/SourceFieldIT.java:19]: Oh, right, checkstyle errors are ignored in IT.

File: build.gradle (hunk 3 above)
Comment (MaxKsyunz) [reply build.gradle:102]: Good catch! I'll remove it -- it does not apply to 2.x branch.

The upstream PR moved that line because the build was picking up dependencies from that random repository instead of sonatype and failing. 
image

It was added when main was [bumped to 3.0](https://github.com/opensearch-project/sql/commit/5dfc2e5ba5d38ce1d0ec35a7904036765e786908).

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by dai-chen — APPROVED** [review]
Thanks for the fix!
